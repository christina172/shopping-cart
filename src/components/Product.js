import cartColor from "./images/cart-trolley-svgrepo-com-color.svg";

function Product(props) {
    let element;
    if (props.product.quantity === 0) {
        element = <img className="cart" src={cartColor} alt="cart" onClick={() => props.addToCart(props.product.name)}></img>
    } else {
        element = <div className="quantity">
            <form onSubmit={props.handleSubmit}>
                <button onClick={() => props.subtract(props.product.name)}>-</button>
                <input type="number" id="quantity" name="quantity" value={props.product.quantity} onChange={(event) => props.handleQuantityChange(event, props.product.name)}></input>
                <button onClick={() => props.add(props.product.name)}>+</button>
            </form>
        </div>
    }
    return (
        <div className="product">
            <div className="product-image">
                <img className="first-image" src={props.product.image1} alt={props.product.name}></img>
                <img className="second-image" src={props.product.image2} alt={props.product.name}></img>
            </div>
            <div className="product-description">
                <div className="product-name">{props.product.name}</div>
                <div className="product-price">{props.product.price} $</div>
                {element}
            </div>
        </div >
    )
};

{/* <button onClick={() => props.addToCart(props.product.name)}> Add to cart</button> */ }

export default Product;