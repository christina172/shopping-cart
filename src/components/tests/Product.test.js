import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import Product from "../Product";

import gammaAquarelleBig1 from "../images/gamma-aquarelle-18-1.jpg";
import gammaAquarelleBig2 from "../images/gamma-aquarelle-18-2.jpg";
import kohInoorAquarellePencilsSmall1 from "../images/koh-i-noor-aquarelle-pencils-24-1.jpeg";
import kohInoorAquarellePencilsSmall2 from "../images/koh-i-noor-aquarelle-pencils-24-2.jpeg";

describe("Product component", () => {
    it("renders Product component", () => {
        const { container } = render(
            <Product key='Watercolour set "Studio", 18 colours, cuvettes (2.6 ml), cardboard box, GAMMA' product={{
                name: 'Watercolour set "Studio", 18 colours, cuvettes (2.6 ml), cardboard box, GAMMA',
                image1: gammaAquarelleBig1,
                image2: gammaAquarelleBig2,
                price: 4.90,
                quantity: 0,
                class: "watercolour-big-gamma"
            }} />
        );
        expect(container).toMatchSnapshot();
    });

    it("renders Product component with input field if quantity is greater than 0", () => {
        render(<Product key='Set of watercolour pencils 3718 FRUIT, 24 colours, cardboard box, KOH-I-NOOR' product={{
            name: `Set of watercolour pencils 3718 FRUIT, 24 colours, cardboard box, KOH-I-NOOR`,
            image1: kohInoorAquarellePencilsSmall1,
            image2: kohInoorAquarellePencilsSmall2,
            price: 17.30,
            quantity: 2,
            class: "watercolour-pencils-small-koh-i-noor"
        }} />)
        expect(screen.getByRole("spinbutton")).toBeInTheDocument();
    });
})