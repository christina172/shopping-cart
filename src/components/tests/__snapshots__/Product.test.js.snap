// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product component renders Product component 1`] = `
<div>
  <div
    class="product"
  >
    <div
      class="product-image"
    >
      <img
        alt="Watercolour set \\"Studio\\", 18 colours, cuvettes (2.6 ml), cardboard box, GAMMA"
        class="first-image"
        src="gamma-aquarelle-18-1.jpg"
      />
      <img
        alt="Watercolour set \\"Studio\\", 18 colours, cuvettes (2.6 ml), cardboard box, GAMMA"
        class="second-image"
        src="gamma-aquarelle-18-2.jpg"
      />
    </div>
    <div
      class="product-description"
    >
      <div
        class="product-name"
      >
        Watercolour set "Studio", 18 colours, cuvettes (2.6 ml), cardboard box, GAMMA
      </div>
      <div
        class="product-price"
      >
        4.9
         $
      </div>
      <img
        alt="cart"
        class="cart"
        src="cart-trolley-svgrepo-com-color.svg"
      />
    </div>
  </div>
</div>
`;
